### ISSUE TITLE
failed to deploy gcs cluster using vagrant.

### ISSUE THREAD

**Issue by ksandha** [gluster/gcs#75]
The cluster fails to deploy using vagrant:- 
```
TASK [GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready] **********
Tuesday 27 November 2018  15:44:42 +0530 (0:00:00.190)       1:08:21.441 ****** 
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (50 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (49 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (48 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (47 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (46 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (45 retries left).




FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (44 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (43 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (42 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (41 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (40 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (39 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (38 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (37 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (36 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (35 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (34 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (33 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (32 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (31 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (30 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (29 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (28 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (27 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (26 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (25 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (24 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (23 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (22 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (21 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (20 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (19 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (18 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (17 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (16 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (15 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (14 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (13 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (12 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (11 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (10 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (9 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (8 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (7 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (6 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (5 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (4 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (3 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (2 retries left).
FAILED - RETRYING: GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready (1 retries left).
fatal: [kube1]: FAILED! => {"attempts": 50, "changed": false, "connection": "close", "content_length": "542", "content_type": "application/json; charset=UTF-8", "cookies": {}, "cookies_string": "", "date": "Tue, 27 Nov 2018 10:23:48 GMT", "json": [{"client-addresses": ["gluster-kube1-0.glusterd2.gcs:24007"], "id": "f71b18e2-badb-44f9-a3cb-ebaab3ea7911", "metadata": {"_zone": "f71b18e2-badb-44f9-a3cb-ebaab3ea7911"}, "name": "gluster-kube1-0", "online": true, "peer-addresses": ["gluster-kube1-0.glusterd2.gcs:24008"], "pid": 26}, {"client-addresses": ["gluster-kube3-0.glusterd2.gcs:24007"], "id": "ff7a5a54-ff8e-49c4-8a09-ac7ee9955787", "metadata": {"_zone": "ff7a5a54-ff8e-49c4-8a09-ac7ee9955787"}, "name": "gluster-kube3-0", "online": true, "peer-addresses": ["gluster-kube3-0.glusterd2.gcs:24008"], "pid": 25}], "msg": "OK (542 bytes)", "redirected": false, "status": 200, "url": "http://10.233.19.39:24007/v1/peers", "x_gluster_cluster_id": "348d6571-88a2-4231-9100-98ef06ad9a9c", "x_gluster_peer_id": "ff7a5a54-ff8e-49c4-8a09-ac7ee9955787", "x_request_id": "7c157d53-b8ed-4551-8a55-977731c39b7d"}
	to retry, use: --limit @/root/gcs-latest/gcs-latest-karan/deploy/vagrant-playbook.retry

PLAY RECAP *********************************************************************
kube1                      : ok=377  changed=114  unreachable=0    failed=1   
kube2                      : ok=4    changed=4    unreachable=0    failed=1   
kube3                      : ok=254  changed=71   unreachable=0    failed=0   

Tuesday 27 November 2018  15:53:48 +0530 (0:09:05.631)       1:17:27.072 ****** 
=============================================================================== 
download : container_download | Download containers if pull is required or told to always pull (all nodes)  2455.31s
GCS | GD2 Cluster | Wait for glusterd2-cluster to become ready -------- 545.63s
GCS | ETCD Operator | Wait for etcd-operator to be available ---------- 433.74s
Install packages ------------------------------------------------------ 153.73s
GCS | ETCD Cluster | Wait for etcd-cluster to become ready ------------ 143.99s
download : container_download | Download containers if pull is required or told to always pull (all nodes) - 127.84s
download : container_download | Download containers if pull is required or told to always pull (all nodes) -- 63.79s
download : container_download | Download containers if pull is required or told to always pull (all nodes) -- 53.10s
download : file_download | Download item ------------------------------- 50.40s
download : container_download | Download containers if pull is required or told to always pull (all nodes) -- 49.67s
download : container_download | Download containers if pull is required or told to always pull (all nodes) -- 37.31s
kubernetes/master : Master | wait for the apiserver to be running ------ 30.71s
etcd : Gen_certs | Write etcd master certs ----------------------------- 18.82s
Wait for host to be available ------------------------------------------ 16.46s
download : container_download | Download containers if pull is required or told to always pull (all nodes) -- 15.67s
GCS Pre | Manifests | Sync GCS manifests ------------------------------- 12.62s
etcd : Configure | Check if etcd cluster is healthy -------------------- 11.26s
etcd : reload etcd ----------------------------------------------------- 10.70s
docker : Docker | pause while Docker restarts -------------------------- 10.23s
kubernetes/master : Master | wait for kube-controller-manager ----------- 9.94s
==> kube3: An error occurred. The error will be shown after all tasks complete.
An error occurred while executing multiple actions in parallel.
Any errors that occurred are shown below.

An error occurred while executing the action on the 'kube3'
machine. Please handle this error then try again:

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
(gcs-venv) [root@rhsqa-virt05 deploy]# 
(gcs-venv) [root@rhsqa-virt05 deploy]# 
(gcs-venv) [root@rhsqa-virt05 deploy]# 
(gcs-venv) [root@rhsqa-virt05 deploy]# 
(gcs-venv) [root@rhsqa-virt05 deploy]# 
(gcs-venv) [root@rhsqa-virt05 deploy]# 
(gcs-venv) [root@rhsqa-virt05 deploy]# 
(gcs-venv) [root@rhsqa-virt05 deploy]# vagrant ssh kube1
[vagrant@kube1 ~]$ 
[vagrant@kube1 ~]$ 
[vagrant@kube1 ~]$ 
[vagrant@kube1 ~]$ 
[vagrant@kube1 ~]$ kubectl get pods -n gcs
NAME                             READY   STATUS    RESTARTS   AGE
etcd-ngw4zkd6g8                  1/1     Running   0          15m
etcd-nv9n225zsd                  1/1     Running   0          13m
etcd-operator-7cb5bd459b-c6zn8   1/1     Running   0          23m
etcd-r68w6jxjrh                  1/1     Running   0          14m
gluster-kube1-0                  1/1     Running   0          13m
gluster-kube2-0                  0/1     Pending   0          13m
gluster-kube3-0                  1/1     Running   0          13m
[vagrant@kube1 ~]$ 
```
The CSI pods aren't even deployed.  and gluster pods going in pending state.

**Comment by ksandha** [comment]
Setup 2 :- 


```
TASK [GCS | ETCD Cluster | Wait for etcd-cluster to become ready] **************
Tuesday 27 November 2018  15:33:29 +0530 (0:00:00.179)       0:36:16.033 ****** 
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (50 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (49 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (48 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (47 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (46 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (45 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (44 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (43 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (42 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (41 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (40 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (39 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (38 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (37 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (36 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (35 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (34 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (33 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (32 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (31 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (30 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (29 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (28 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (27 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (26 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (25 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (24 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (23 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (22 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (21 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (20 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (19 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (18 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (17 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (16 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (15 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (14 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (13 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (12 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (11 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (10 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (9 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (8 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (7 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (6 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (5 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (4 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (3 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (2 retries left).
FAILED - RETRYING: GCS | ETCD Cluster | Wait for etcd-cluster to become ready (1 retries left).
fatal: [kube1]: FAILED! => {"attempts": 50, "changed": false, "content": "", "msg": "Status code was -1 and not [200]: Request failed: <urlopen error [Errno 113] No route to host>", "redirected": false, "status": -1, "url": "http://10.233.60.17:2379/v2/members"}
	to retry, use: --limit @/root/gcs-latest/gcs-latest-karan2/deploy/vagrant-playbook.retry

PLAY RECAP *********************************************************************
kube1                      : ok=365  changed=112  unreachable=0    failed=1   
kube2                      : ok=292  changed=91   unreachable=0    failed=0   
kube3                      : ok=253  changed=71   unreachable=0    failed=0   

Tuesday 27 November 2018  15:43:59 +0530 (0:10:30.643)       0:46:46.677 ****** 
=============================================================================== 
GCS | ETCD Cluster | Wait for etcd-cluster to become ready ------------ 630.64s
download : container_download | Download containers if pull is required or told to always pull (all nodes) - 389.77s
download : container_download | Download containers if pull is required or told to always pull (all nodes) - 376.40s
Install packages ------------------------------------------------------ 264.78s
download : container_download | Download containers if pull is required or told to always pull (all nodes) - 192.70s
download : file_download | Download item ------------------------------ 148.51s
GCS | ETCD Operator | Wait for etcd-operator to be available ----------- 95.88s
download : container_download | Download containers if pull is required or told to always pull (all nodes) -- 60.78s
download : container_download | Download containers if pull is required or told to always pull (all nodes) -- 46.46s
kubernetes/master : Master | wait for the apiserver to be running ------ 32.00s
download : container_download | Download containers if pull is required or told to always pull (all nodes) -- 29.60s
Wait for host to be available ------------------------------------------ 23.29s
download : container_download | Download containers if pull is required or told to always pull (all nodes) -- 20.53s
etcd : Gen_certs | Write etcd master certs ----------------------------- 20.32s
GCS Pre | Manifests | Sync GCS manifests ------------------------------- 11.84s
kubernetes/master : Master | wait for kube-controller-manager ---------- 11.26s
etcd : reload etcd ----------------------------------------------------- 10.88s
docker : Docker | pause while Docker restarts -------------------------- 10.21s
download : container_download | Download containers if pull is required or told to always pull (all nodes) --- 8.87s
etcd : wait for etcd up ------------------------------------------------- 7.51s
==> kube3: An error occurred. The error will be shown after all tasks complete.
An error occurred while executing multiple actions in parallel.
Any errors that occurred are shown below.

An error occurred while executing the action on the 'kube3'
machine. Please handle this error then try again:

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
(gcs-venv) [root@rhsqa-virt05 deploy]# 
(gcs-venv) [root@rhsqa-virt05 deploy]# 
(gcs-venv) [root@rhsqa-virt05 deploy]# 
(gcs-venv) [root@rhsqa-virt05 deploy]# 
(gcs-venv) [root@rhsqa-virt05 deploy]# 
(gcs-venv) [root@rhsqa-virt05 deploy]# vagrant ssh kube1
[vagrant@kube1 ~]$ 
[vagrant@kube1 ~]$ kubectl get pods -n gcs
NAME                             READY   STATUS      RESTARTS   AGE
etcd-6mwvjrc8mt                  0/1     Error       0          31m
etcd-operator-7cb5bd459b-xk68n   1/1     Running     0          33m
etcd-tznzpbsxx5                  0/1     Completed   0          32m
[vagrant@kube1 ~]$ 

```

**Comment by ksandha** [comment]
Closing this issue as the deployment is working with today's nightly.
